crates/nu-std/lib/dirs.nu

```diff
@@ -74,7 +76,29 @@ export def-env "dirs show" [] {
 }
 
 export alias shells = dirs show
-export alias g = dirs show
+
+export def-env "dirs goto" [shell?: int] {
+    if $shell == null {
+        return (dirs show)
+    }
+
+    if $shell < 0 or $shell >= ($env.DIRS_LIST | length) {
+        let span = (metadata $shell | get span)
+        error make {
+            msg: $"(ansi red_bold)invalid_shell_index(ansi reset)"
+            label: {
+                text: $"`shell` should be between 0 and (($env.DIRS_LIST | length) - 1)"
+                start: $span.start
+                end: $span.end
+            }
+        }
+    }
+    let-env DIRS_POSITION = $shell
+
+    cd ($env.DIRS_LIST | get $env.DIRS_POSITION)
```

Comment:
I'm not sure if all this is necessary when we already have this type of checking in `_fetch`, just done differently. Am I missing something? I do like checking for null and doing `dirs show` if it's null.

Reply:
it's not really the same
- `_fetch` changes the current shell by an offset, e.g. by `+2` or `-3` and then wrapping around with `mod` the size of the current list of shells
- built-in `g` does not go by relative offset but rather by absolute index, hence the new `dirs goto` which does this exactly

:bulb: as stated in the help of built-in `g`
> Switch to a given shell, or list all shells if no given shell number.
>
and
> shell_number <one_of(int, string)>: shell number to change to (optional)
>
whereas `_fetch` uses an `offset: int`

does that answer your concern? :blush:

Reply:
Yes, it does. Thanks for explaining it.

Reply:
:relieved: